spiceaidocs/content/en/reference/Secrets/_index.md

```diff
@@ -0,0 +1,100 @@
+---
+type: docs
+title: "Spice Secrets"
+linkTitle: "Spice Secrets"
+description: ''
+weight: 70
+---
+
+## Secret
+
+A Secret is an object used to store sensitive data, like a passwords, a tokens, or keys.
+
+## Secret Stores
+
+The secret store is a location where the `secret` objects are stored.
+
+Spice.ai supports four types of secret stores: `file`, `env`, `kubernetes` and `keyring`. The type of secret store is specified in the `store` field of the `secrets` section in the Spicepod manifest.
+
+### File Secret Store
+
+Default secret store uses a file located at `~/.spice/auth`.
+The Spice.ai CLI offers the `spice login` command to streamline credential storage. When logging into Spice.ai, it automatically saves credentials in the secret store file under `spiceai` secret.
+
+### Environment Secret Store
+
+The `env` store type allows Spice.ai to read secrets from environment variables. The environment variables should be formatted like `SPICE_SECRET_<secret-name>_<secret-value-key>`. 
+
+All variables with the same prefix `SPICE_SECRET_<secret-name>` are combined into a single secret. This allows you to group related secret values under a single secret name.
+
+**Example**
+
+```yaml
+secrets:
+  store: env
+```
+
+Setting `spiceai` secret with spice.ai API key in `key` secret value:
+
+```bash
+SPICE_SECRET_SPICEAI_KEY="343533|**************" \
+  spice run
+```
+
+### Kubernetes Secret Store
+
+The `kubernetes` store type allows Spice.ai to read Kubernetes secrets.
+
+```yaml
+secrets:
+  store: kubernetes
+```
+
+Note: This method requires the Kubernetes service account, which is running the Spice.ai pod, to have extended roles for secrets API access. Make sure to configure this service account with the necessary permissions to read secrets from the Kubernetes API.
+
+Example of Kubernetes role configuration for a custom service account:
+
+```yaml
+kind: Role
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: spiced-account-role
+rules:
+  - apiGroups: [""]
+    resources: ["secrets"]
+    verbs: ["get"]
+```
+
+### Keyring Secret Store
+
+> ⚠️ Keyring store can be enabled by building [spiced](https://github.com/spiceai/spiceai/tree/trunk/bin/spiced) with `keyring-secret-store` cargo feature.
```

Comment:
```suggestion
> ⚠️ Keyring store can be enabled by building [spiced](https://github.com/spiceai/spiceai/tree/trunk/bin/spiced) with `keyring-secret-store` cargo feature.
```

This should be enabled by default

Reply:
Good point, removed that note. Keyring enabled by default in https://github.com/spiceai/spiceai/pull/800